programs/aes/aescrypt2.c

```diff
@@ -447,8 +452,8 @@ int main( int argc, char *argv[] )
     mbedtls_aes_free( &aes_ctx );
     mbedtls_md_free( &sha_ctx );
 
-    if( ret != 0 && ret != 1)
-        ret = 1;
+    if( ret != MBEDTLS_EXIT_SUCCESS && ret != MBEDTLS_EXIT_FAILURE )
```

Comment:
It probably doesn't matter, but wouldn't be `if( ret != MBEDTLS_EXIT_SUCCESS )` sufficient?